octoprint_octorelay/__init__.py

```diff
@@ -211,19 +232,21 @@ def handler(task: Task):
             if same_subject and not_exception and same_target and same_owner:
                 try:
                     task.cancel_timer()
-                    self._logger.info(f"cancelled {task}")
+                    self._logger.info(f"Cancelled the task: {task}")
                 except Exception as exception:
-                    self._logger.warn(f"failed to cancel {task}, reason: {exception}")
+                    self._logger.warn(f"Failed to cancel {task}, reason: {exception}")
                 return False # exclude
             return True # include
         self.tasks = list(filter(handler, self.tasks))
+        self._logger.debug("The cancelled tasks removed from the registry")
 
     def run_system_command(self, cmd):
         if cmd:
-            self._logger.info(f"OctoRelay runs system command: {cmd}")
+            self._logger.debug(f"Running the system command: {cmd}")
             os.system(cmd)
 
     def get_upcoming_tasks(self, subjects):
+        self._logger.debug(f"Finding the upcoming tasks for {list(subjects)}")
```

Comment:
🚀 3.8.0rc2

Somehow making the `list()` corrupts the workflow of this function.
Perhaps it mutates the `subjects` in-place.

I checked that removing it — fixes it.

```suggestion
        self._logger.debug("Finding the upcoming tasks")
```